src/main/java/org/wso2/extension/siddhi/window/uniquetimebatch/UniqueTimeBatchWindowProcessor.java

```diff
@@ -0,0 +1,379 @@
+                    timeInMilliSeconds = (Integer)
+                            ((ConstantExpressionExecutor) attributeExpressionExecutors[1]).getValue();
+                } else if (attributeExpressionExecutors[1].getReturnType() == Attribute.Type.LONG) {
+                    timeInMilliSeconds = (Long)
+                            ((ConstantExpressionExecutor) attributeExpressionExecutors[1]).getValue();
+                } else {
+                    throw new ExecutionPlanValidationException("UniqueTimeBatch window's parameter time should be either" +
+                            " int or long, but found " + attributeExpressionExecutors[1].getReturnType());
+                }
+            } else {
+                throw new ExecutionPlanValidationException("Unique Time Batch window should have constant " +
+                        "for time parameter but found a dynamic attribute "
+                        + attributeExpressionExecutors[1].getClass().getCanonicalName());
+            }
+        } else if (attributeExpressionExecutors.length == 3) {
+            uniqueKey = (VariableExpressionExecutor) attributeExpressionExecutors[0];
+            if (attributeExpressionExecutors[1] instanceof ConstantExpressionExecutor) {
+                if (attributeExpressionExecutors[1].getReturnType() == Attribute.Type.INT) {
+                    timeInMilliSeconds = (Integer)
+                            ((ConstantExpressionExecutor) attributeExpressionExecutors[1]).getValue();
+                } else if (attributeExpressionExecutors[1].getReturnType() == Attribute.Type.LONG) {
+                    timeInMilliSeconds = (Long)
+                            ((ConstantExpressionExecutor) attributeExpressionExecutors[1]).getValue();
+                } else {
+                    throw new ExecutionPlanValidationException("UniqueTimeBatch window's parameter time should be either" +
+                            " int or long, but found " + attributeExpressionExecutors[1].getReturnType());
+                }
+            } else {
+                throw new ExecutionPlanValidationException("Unique Time Batch window should have constant " +
+                        "for time parameter but found a dynamic attribute "
+                        + attributeExpressionExecutors[1].getClass().getCanonicalName());
+            }
+            // isStartTimeEnabled used to set start time
+            if (attributeExpressionExecutors[2].getReturnType() == Attribute.Type.BOOL) {
+                this.isFirstUniqueEnabled = (boolean) (((ConstantExpressionExecutor)
+                        attributeExpressionExecutors[2]).getValue());
+
+            } else if (attributeExpressionExecutors[2].getReturnType() == Attribute.Type.INT) {
+                isStartTimeEnabled = true;
+                startTime = Integer.parseInt(String
+                        .valueOf(((ConstantExpressionExecutor) attributeExpressionExecutors[2]).getValue()));
+            } else {
```

Comment:
We should check whether user actually gives a LONG.
`else if (attributeExpressionExecutors[2].getReturnType() == Attribute.Type.LONG) ` 

and then let it be parsed to long. 
Else we should throw an error like "Expected either boolean, int or long type for UniqueTimeBatch window's third parameter, but found " + attributeExpressionExecutors[2].getReturnType()"